### ISSUE TITLE
No uniquely matching class member, but no candidates are printed.

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9148]
**Describe the bug**
We get 100+ warnings of the type:
```
/opt/root_src/io/sql/src/TSQLFile.cxx:2668: warning: no uniquely matching class member found for 
  void TSQLFile::Streamer(TBuffer &b)
```
but then no "candidates" are proposed.

**Expected behavior**
Either the adjective "uniquely" should be removed, or the "potential candidates", if any, should be listed.

**Screenshots**
Log is attached: [consoleText.txt.zip](https://github.com/doxygen/doxygen/files/8118260/consoleText.txt.zip)

**To Reproduce**
I couldn't yet achieve a "small" reproducer, somehow it only appears in the "big build".

**Version**
1.9.4 (a85850bf5b32843d6cde6b9a25b147c9a40bbc8a*)

**Comment by albert-github** [comment]
You didn't provide the used Doxyfile but from the other questions I see a repeating pattern, so maybe a bit of a wild guess, have a look at:
```
IGNORE_PREFIX          = T
EXCLUDE_PATTERNS    = ....
```

**Comment by albert-github** [comment]
I assume that the normal declaration is from `core/base/inc/Rtypes.h`.
What are the settings for `virtual_keyword` and for `overrd`?

Might be useful yo set `INPUT` to `INPUT=io/sql/src/TSQLFile.cxx` and try to run just the doxygen paryt as `doxygen -d preprocessor`.

**Comment by ferdymercury** [comment]
Thanks for the hint. Attached is the preprocessor output with just the TSQLFile.cxx and .h as input.
[preproc.txt.zip](https://github.com/doxygen/doxygen/files/8119323/preproc.txt.zip)

overrd = override, and virtual_keyword is empty, because:
`ClassDefOverride(TSQLFile, 1)` calls (via Rtypes.h) the following:
`_ClassDefOutline_(name,id,,override)`
which has one empty argument (the third, i.e. the virtual_keyword), maybe that's confusing for doxygen?

This one then declares the Streamer function as:
`virtual_keyword void Streamer(TBuffer&) overrd;`
=>
`void Streamer(TBuffer&) override;`

We do see issues with anonymous namespaces in the full log, but not in this particular class, as far as I can evaluate.

Thanks again!

**Comment by ferdymercury** [comment]
Ok, I think I have found the culprit:
`#include TFile.h: not found! skipping...`

which means I should add this file to the include_path, too.

However, I think it would make more sense to remove the "uniquely" from the warning message.

**Comment by ferdymercury** [comment]
Ok, I managed to create a (small) reproducer. Here it is:
[uniquely.zip](https://github.com/doxygen/doxygen/files/8119477/uniquely.zip)

Here are the different output errors.

If I let everything as in the zip:
```
TSQLFile.cxx:2668: warning: no uniquely matching class member found for 
  void TSQLFile::Streamer(TBuffer &b)
```

If add subfolder to the include_path in the Doxyfile, no errors are found.

If I remove Rtypes.h, the output error makes much more sense:
`streamer/TSQLFile.cxx:2668: warning: documented symbol 'void TSQLFile::Streamer' was not declared or defined.`

**Comment by ferdymercury** [comment]
"needinfo" label can be removed? Or you need some more info from my side? Thanks!

**Comment by ferdymercury** [comment]
I just proposed a patch. https://github.com/doxygen/doxygen/pull/9293

With it, this how the output looks like now. No longer empty:
```

/tmp/uniquely/TSQLFile.cxx:2668: warning: no uniquely matching class member found for 
  void TSQLFile::Streamer(TBuffer &b)
Possible candidates:
  'void TFile::Streamer(TBuffer &) override' at line 341 of file /tmp/uniquely/subfolder/TFile.h
  'void TDirectory::Streamer(TBuffer &) override' at line 299 of file /tmp/uniquely/TDirectory.h
  'void TDirectoryFile::Streamer(TBuffer &) override' at line 130 of file /tmp/uniquely/TDirectoryFile.h
  'virtual void TNamed::Streamer(TBuffer &)' at line 58 of file /tmp/uniquely/TNamed.h
  'virtual void TObject::Streamer(TBuffer &)' at line 230 of file /tmp/uniquely/TObject.h
```

**Comment by ferdymercury** [comment]
@albert-github can you add the fixed-not-released label ? Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
